### ISSUE TITLE
Simplified output formats: Markdown and JSON

### ISSUE THREAD

**Issue by lfoppiano** [kermitt2/grobid#1306]
For whom that wants to use a simpler format for the fulltext that look less difficult than XML we could create a couple of transformation XSLT that would output Markdown and JSON. 

Since those formats are very simple they will be designed for fulltext processing rather than rich data structuring (as it is now the XML). 

XML will be the default format specified without parameters.

**Comment by lfoppiano** [comment]
@LaurentRomary do you think would be difficult to write two XSLT transformations sheets from the TEI to Markdown and JSON? 

Markdown could be something like 

```markdown 

Title 

Authors.... 

Affiliations .... 

DOI: 

text with <head> becoming heading information and text streamed out with an empty line between each paragraphs

```

For JSON I have a couple of examples that could work.

**Comment by laurentromary** [comment]
Yes, I can take care of it during the summer. Send me your example and I'll start ASAP.

**Comment by lfoppiano** [comment]
@laurentromary @Madjakul is preparing a more systematic conversion table to cover more cases. Meanwhile I'm sending you an example with more details we can start iterating over: 

PDF: [NeurIPS-2024-gita-graph-to-visual-and-textual-integration-for-vision-language-graph-reasoning-Paper-Conference.pdf](https://github.com/user-attachments/files/21019907/NeurIPS-2024-gita-graph-to-visual-and-textual-integration-for-vision-language-graph-reasoning-Paper-Conference.pdf)

TEI: [NeurIPS-2024-gita-graph-to-visual-and-textual-integration-for-vision-language-graph-reasoning-Paper-Conference.pdf.tei.xml.zip](https://github.com/user-attachments/files/21020345/NeurIPS-2024-gita-graph-to-visual-and-textual-integration-for-vision-language-graph-reasoning-Paper-Conference.pdf.tei.xml.zip)

-----


# GITA: Graph to Visual and Textual Integration for Vision-Language Graph Reasoning

Yanbin Wei, Shuai Fu, Weisen Jiang, Qi Wu, James T Kwok, Yu Zhang

DOI (when available)

Large Language Models (LLMs) are increasingly used for various tasks with graph structures. Though LLMs can process graph information in the textual format, they overlook the rich vision modality, which is an intuitive way for humans to comprehend structural information and conduct general graph reasoning. The potential benefits and capabilities of representing graph structures as visual images (i.e., visual graph) are still unexplored. To fill the gap, we innovatively propose an end-to-end framework, called Graph to vIsual and Textual IntegrAtion (GITA), which incorporates visual graphs into general graph reasoning. Besides, we construct the Graph-based Vision-Language Question Answering (GVLQA) dataset from existing graph data, which is the first vision-language dataset for general graph reasoning. Extensive experiments on the GVLQA dataset and five real-world datasets show that GITA outperforms mainstream LLMs on general graph reasoning. Moreover, experimental results demonstrate the effectiveness of the layout augmentation on visual graphs and pretraining on the GVLQA dataset.

## Introduction

Graph reasoning tasks are pivotal in domains such as recommendation systems [\[25,](b24) [60\]](b25), social network analysis [7](b6), and knowledge graph reasoning [72](#b71). Various architectures have been developed, from shallow embedding methods [6](b5) to advanced Graph Neural Networks (GNNs) [37](b36) and graph Transformers [71](b70). While these models excel in graph reasoning tasks, they often lack generalizability, flexibility, and user-friendliness. Achieving good performance with these models typically requires domain-specific tuning, which limits their abilities to generalize across different domains. Additionally, these models struggle to handle diverse tasks with the same architecture. Each task often requires a specialized design, including task-specific data processing and decoder, leading to limited flexibility. Lastly, unlike the Large Language Models (LLMs) that can engage in conversations with users, these models are less explainable and user-friendly.

In contrast, LLMs have shown great generalization capabilities across a wide variety of reasoning tasks [\[61](#b60), [67](#b66), [74](#b73), [32](#b31), [33\]](#b32) by encapsulating task-specific demands within a cohesive and interpretable mechanism - text prompts, under a unified architecture with minimal domain-specific adjustments. These advantages have sparked investigations into the potential of LLMs for graph reasoning. Recent developments lend credence to the notion that LLMs can indeed interpret and manipulate graph-structured data through textual representations. For example, InstructGLM [66](#b65), GPT4Graph [21](#b20), and LLMtoGraph [44](#b43) convert graphs into textual descriptions and then use these descriptions paired with queries to enable LLMs to generate accurate responses for graph reasoning tasks. Furthermore, the introduction of benchmarks such as GraphQA [14](#b13) and NLGraph [59](#b58) is a testament to the growing interest in evaluating LLMs' effectiveness on graph reasoning tasks framed in natural languages.

Despite the development of numerous methods and benchmarks for graph reasoning on LLMs, they often overlook the valuable vision modality, which is a natural means for humans to comprehend structural information and has demonstrated its success in various visual reasoning scenarios [\[30](#b29), [68](#b67), [69](#b68), [34](#b33), [3](#b2), [54\]](#b53). Consequently, the following questions arise: [1](#b0) Can incorporating visual information be beneficial for general graph reasoning scenarios? (2) If so, how can we effectively integrate the vision modality into graph reasoning? To the best of our knowledge, these questions remain unexplored.

## Related Work

[....]

| Models              | Connect Cycle |       | TS    | SP    | MaxFlow BGM | HP    | Avg              |
|---------------------|---------------|-------|-------|-------|--------------|-------|------------------|
|                     |               |       |       |       |   Zero-shot           |       |                  |
| LLaMA2-7B          | 50.06         | 49.43 | 0.00  | 0.00  | 0.00        | 0.00  | 0.00 14.21       |
| Vicuna-7B          | 50.06         | 49.43 | 0.00  | 0.00  | 0.00        | 0.00  | 0.00 14.21       |
| GITA-7B (VO)       | 50.06         | 50.33 | 0.00  | 0.00  | 0.00        | 0.00  | 0.00 14.34       |
| GITA-7B            | 50.06         | 49.43 | 0.00  | 0.00  | 0.00        | 0.00  | 0.00 14.21       |
| GPT-4 Turbo        | 76.70         | 49.51 | 19.59 | 35.35 | 6.89        | 42.11 | 47.04 39.60      |
| GITA-ZS (VO)       | 57.76         | 63.34 | 5.34  | 4.88  | 1.59        | 46.60 | 10.74 27.18      |
| GITA-ZS            | 82.58         | 51.46 | 19.71 | 37.69 | 6.00        | 52.21 | 50.00 42.81      |
|                     |               |       |       |       |              |       |                  |
| LLaMA2-7B          | 97.33         | 94.63 | 33.26 | 26.01 | 9.56        | 90.86 | 23.95 53.66      |
| Vicuna-7B          | 97.58         | 95.04 | 34.46 | 25.98 | 9.33        | 91.04 | 25.55 54.15      |
| GITA-7B (VO)       | 59.97         | 96.34 | 13.30 | 5.72  | 2.89        | 93.01 | 1.11 38.91       |
| GITA-7B            | 98.95         | 96.67 | 41.12 | 32.15 | 20.00       | 93.19 | 29.26 58.76      |
| LLaMA2-13B         | 98.79         | 93.36 | 33.83 | 27.93 | 12.22       | 91.34 | 33.46 55.85      |
| Vicuna-13B         | 99.35         | 94.39 | 36.73 | 28.53 | 11.34       | 92.65 | 34.81 56.83      |
| GITA-13B (VO)      | 58.00         | 96.91 | 14.45 | 5.72  | 4.89        | 93.19 | 1.85 39.29       |
| GITA-13B           | 99.14         | 95.60 | 38.69 | 40.47 | 20.66       | 92.12 | 33.33 60.00      |

**Table 2:** Accuracy (%) comparisons across GVLQA subsets using GITA-7B (VO). ↑ denotes dramatic performance improvement.

[...]

**Figure 1:** The architecture of the GITA framework with comparison to existing LLM solution.

**Figure 2:** A comparative case study of graph representation in vision and text modalities. All methods are trained on the GVLQA-BASE dataset.

[...]

# References

[b0] **Lada, A., Adamic, Natalie, & Glance.** (2005). *The political blogosphere and the 2004 US election: divided they blog*. In *Proceedings of the 3rd International Workshop on Link Discovery* (pp. 36-43). The 3rd International Workshop on Link Discovery.

[...]

-------

This is a rough example: 
 - bibliographic data are only provided as title, authors, DOI
 - inline references are transformed in links with the target being the link destination 
 - paragraphs are separated by an empty line, head of section are transformed as sub-titles 
 - references are streamed as follow: authors, year, title, and other references to follow 
 - tables are transformed with `<row>` being a row in markdown and a `<cell>` being the item in the row 

example of tables 

```
<table>
                    <row>
                        <cell>Models</cell>
                        <cell cols="2">Connect Cycle</cell>
                        <cell>TS</cell>
                        <cell>SP</cell>
                        <cell cols="2">MaxFlow BGM</cell>
                        <cell>HP</cell>
                        <cell>Avg</cell>
                    </row>
                    <row>
                        <cell/>
                        <cell/>
                        <cell/>
                        <cell cols="2">Zero-shot</cell>
                        <cell/>
                        <cell/>
                    </row>
```
should become 

| Models              | Connect Cycle |       | TS    | SP    | MaxFlow BGM | HP    | Avg              |
|---------------------|---------------|-------|-------|-------|--------------|-------|------------------|
|                     |               |       |       |       |   Zero-shot           |       |                  |

**Comment by lfoppiano** [comment]
After a discussion with OpenAlex, I came to realise that we don't need to pass through all the models we normally do when we output Markdown, probably only segmentation, fulltext, and everything else can be streamed out, at least to start with.
